partner_address_version/models/res_partner.py

```diff
@@ -13,54 +13,93 @@ class ResPartner(models.Model):
     version_hash = fields.Char(readonly=True, copy=False)
     date_version = fields.Datetime(string='Date version', readonly=True)
 
-    @api.multi
-    def get_address_version(self):
-        """
-        Get a versioned partner corresponding to the partner fields.
-        If no versioned partner exists, create a new one.
-        """
-        self.ensure_one()
-        version_hash = self.get_version_hash()
-        versioned_partner = self.with_context(active_test=False).search(
-            [('version_hash', '=', version_hash)])
-        if versioned_partner:
-            return versioned_partner
-        default = {
-            'active': False,
-            'version_hash': version_hash,
-            'parent_id': self.parent_id and self.parent_id.id or self.id,
-            'date_version': fields.Datetime.now(),
-        }
-        versioned_partner = self.copy(default=default)
-        return versioned_partner
-
     def get_version_fields(self):
+        # deprecated uses _version_fields instead
+        return self._version_fields()
+
+    def _version_fields(self):
         return [
             'name', 'street', 'street2', 'zip', 'city', 'country_id',
             'parent_id'
         ]
 
     def get_version_hash(self):
-        version_fields = self.get_version_fields()
+        # deprecated uses _version_hash instead
```

Comment:
if you want to support backward compat  you should log a deprecation warning :wink: